nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -312,319 +316,2021 @@ impl DataStore {
+                    },
+                    None => {
+                        error!(opctx.log, "{message}"; "error" => ?e);
+                        Error::internal_error("error while adding prefix to allowed import list")
+                    },
+                }
+            })
+    }
+
+    pub async fn switch_port_configuration_bgp_peer_allow_import_remove(
+        &self,
+        opctx: &OpContext,
+        configuration: NameOrId,
+        bgp_peer: IpAddr,
+        prefix: AllowedPrefixAddRemove,
+    ) -> DeleteResult {
+        let conn = self.pool_connection_authorized(opctx).await?;
+        let err = OptionalError::new();
+
+        self.transaction_retry_wrapper(
+            "switch_port_configuration_bgp_peer_allow_import_remove",
+        )
+            .transaction(&conn, |conn| {
+                let parent_configuration = configuration.clone();
+                let settings_to_remove = prefix.clone();
+                let err = err.clone();
+
+                async move {
+                    // resolve id of port_settings record
+                    let port_settings_id = switch_port_configuration_id(&conn, parent_configuration.clone())
+                        .await
+                        .map_err(|e: diesel::result::Error| {
+                            match e {
+                                diesel::result::Error::NotFound => {
+                                    err.bail(
+                                        Error::non_resourcetype_not_found(
+                                            format!("unable to lookup configuration with identifier {parent_configuration}")
+                                        )
+                                    )
+                                },
+                                _ => {
+                                    err.bail(Error::internal_error(
+                                        "error while looking up configuration for interface bgp peer"
+                                    ))
+                                },
+                            }
+                        })?;
+
+                    // delete allowed import
+                    // PRIMARY KEY (port_settings_id, interface_name, addr, prefix)
+                    use db::schema::switch_port_settings_bgp_peer_config_allow_import as allow_import;
+                    diesel::delete(allow_import::table)
+                        .filter(allow_import::port_settings_id.eq(port_settings_id))
+                        .filter(allow_import::interface_name.eq(settings_to_remove.interface.to_string()))
+                        .filter(allow_import::addr.eq(IpNetwork::from(bgp_peer)))
+                        .filter(allow_import::prefix.eq(IpNetwork::from(settings_to_remove.prefix)))
+                        .execute_async(&conn)
+                        .await
+                        .map_err(|e| {
+                            let message = "error while deleting import list entry for bgp peer";
```

Comment:
Here and in line 1948, it references "entry", but in line 1872, it references "prefix".

Check other places as needed, including same for export (vs import).